### PR TITLE
SECD-598 Add Postgres-based implementation of PhysicalAssetStorer interface

### PR DESCRIPTION
**Pull request by bkrebsbach**
Add Postgres-based implementation of PhysicalAssetStorer interface
* Add a PostgresPhysicalAssetStorer to persist physical assets to Postgres
  * Inserts customers, subnets, and devices
  * Add mocked unit tests
* Add PostgresPhysicalAssetStorer to sync handler in main
* Update dependencies

Modify integration tests to run in parallel
* Update PostgresDB SQLDB implementation to gracefully handle multiple instances of ipam-facade service creating database/tables concurrently
* Remove shared state from integration tests in favor of per-test DB connections
  * Configure a PostgresDB, PostgresPhysicalAssetFetcher, and PostgresPhysicalAssetStorer per-test
* Add t.Parallel() flag to all integration tests
* Replace usage of custom "insertX" test functions with PostgresPhysicalAssetStorer to better simulate production code path
* Use require.Nil/require.Error for error handling
* Use random identifiers per test to prevent collision
* Drop unused test functions

### CODE REVIEW

File: pkg/sqldb/postgres.go [hunk 1]
```diff
@@ -84,7 +98,22 @@ func (db *PostgresDB) Init(ctx context.Context, host, port, username, password,
...
+			// is ignored.
+			pqErr := err.(*pq.Error)
+			if pqErr.Code != pqUniqueViolationErrCode {
+				initerr = pqErr
+				return
+			}
+		default:
+			initerr = err
+			return // from the unnamed once.Do function
+		}
```
Comment (faldridge) [inline pkg/sqldb/postgres.go]: question: when both unique violation codepaths are hit, do the tests still pass?

File: pkg/sqldb/postgres_test.go [hunk 2]
```diff
@@ -24,16 +24,14 @@ func TestFetchPhysicalAssetsErrors(t *testing.T) {
...
+		postgresConfig.Username, postgresConfig.Password, postgresConfig.DatabaseName),
+		"DB.Init should have returned a non-nil error")
 }
 
 func TestDoesDBExistTrue(t *testing.T) {
 	mockdb, mock, err := sqlmock.New()
-	if err != nil {
-		t.Fatalf("an error '%s' was not expected when opening a stub database connection", err)
-	}
+	require.Nil(t, err, "an error '%s' was not expected when opening a stub database connection")
```
Comment (faldridge) [inline pkg/sqldb/postgres_test.go]: nit: to take advantage of string interpolation with `require.Nil`, you have to pass the arguments to you wish to interpolate into the string to the `Nil` function. Example: https://play.golang.org/p/HqTYpnKFpmH

File: tests/postgres_test.go [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+		SubnetID:      int64(subnetID),
+		CustomerID:    int64(customerID2),
+	}
+
+	require.Equal(t, expected, asset)
+}
+
+// TestOverlappingSubnet verifies that a query for an IP address will return the
+// most specific subnet that matches, as measured by the subnet's netmask length
+func TestOverlappingSubnet(t *testing.T) {
```
Comment (faldridge) [inline tests/postgres_test.go]: I really like the way you've cleaned up these tests. Thanks!!

File: pkg/sqldb/postgres.go (hunk 1 above)
Comment (bkrebsbach) [reply pkg/sqldb/postgres.go]: Yes, that's what I see in my local testing

File: pkg/sqldb/postgres_test.go (hunk 2 above)
Comment (bkrebsbach) [reply pkg/sqldb/postgres_test.go]: 🤦‍♂ will fix

**Pull request merged**